gapic/src/main/com/google/gapid/views/ProfileView.java

```diff
@@ -343,12 +372,28 @@ protected GpuSliceTrack(long trackId, List<Service.ProfilingData.GpuSlices.Slice
         });
       }
 
-      private static Slices toSlices(Service.ProfilingData.GpuSlices.Slice serverSlice) {
-        return new Slices(Lists.newArrayList(serverSlice), "GPU Queue Events");
+      private static GpuSlices toSlices(Service.ProfilingData.GpuSlices.Slice serverSlice) {
+        return new GpuSlices(Lists.newArrayList(serverSlice), "GPU Queue Events");
+      }
+
+      private static GpuSlices toSlices(List<Service.ProfilingData.GpuSlices.Slice> serverSlices) {
+        return new GpuSlices(serverSlices, "GPU Queue Events");
       }
 
-      private static Slices toSlices(List<Service.ProfilingData.GpuSlices.Slice> serverSlices) {
-        return new Slices(serverSlices, "GPU Queue Events");
+      private static class GpuSlices extends Slices {
+        public final List<Long> groupIds = Lists.newArrayList();
+
+        public GpuSlices(List<Service.ProfilingData.GpuSlices.Slice> serverSlices, String title) {
+          super(serverSlices, title);
```

Comment:
Please remove the constructor that takes the proto slices from the superclass and move its logic here.